init.lua

```diff
@@ -177,53 +177,64 @@ function ranks.update_nametag(player)
 	end
 end
 
--- [function] Set player rank
+-- [function] Set player rank TODO
 function ranks.set_rank(player, rank)
+	local name = player
+
 	if type(player) == "string" then
 		player = minetest.get_player_by_name(player)
 	end
 
 	if registered[rank] then
+
+		storage:set_string(name, rank)
+
 		-- Set attribute
-		player:set_attribute("ranks:rank", rank)
-		-- Update nametag
-		ranks.update_nametag(player)
-		-- Update privileges
-		ranks.update_privs(player)
+		if minetest.get_player_by_name(name) then
+			player:set_attribute("ranks:rank", rank)
+			-- Update nametag
+			ranks.update_nametag(player)
+			-- Update privileges
+			ranks.update_privs(player)
+		end
 
 		return true
 	end
 end
 
 -- [function] Remove rank from player
 function ranks.remove_rank(player)
+	local name = player
+
 	if type(player) == "string" then
 		player = minetest.get_player_by_name(player)
 	end
```

Comment:
As mentioned in the cases of `ranks.set_rank` and `ranks.get_rank`, this should be modified to ensure that the content of `player` is always a string.